### DISCUSSION TITLE
How to access to Cockpit via domain instead of IP address

### DISCUSSION THREAD

**Discussion by maximehamou** [cockpit-project/cockpit#19644] [Q&A] [answered]
### Explain what happens

Hello,
I would like to know how to access to Cockpit panel via a domain name instead of the IP address of the machine, using Apache on Debian 11.
Thank you for your help!

### Version of Cockpit

304

### Where is the problem in Cockpit?

Unknown or not applicable

### Server operating system

Debian

### Server operating system version

11

### What browsers are you using?

Chrome, Safari macOS, Safari on iPhone

**Comment by martinpitt** [comment]
That's not really a cockpit specific question. Do you already have a domain name on your server? If it resolves to the correct IP, what doesn't work? If you don't have a domain name, you need to set one up (this is not trivial, but also this bug tracker isn't the place to give that kind of support).

**Reply by maximehamou** [reply]
@martinpitt Yes I have a domain name configured (I already have 3 website on my VPS). But when I try to configure the redirection in the vhost (with cockpit.domain.fr) from / to http://localhost:9090/, it work for the login page, but not after (it throws an error, but when I go to the logs, there is nothing). Everything is working perfectly with https://my-ip:9090/.

The error in the console from cockpit.domain.fr :
```
[Error] WebSocket connection to 'wss://cockpit.domain.fr/cockpit/socket' failed: There was a bad response from the server.
[Warning] transport closed: disconnected (index.js, line 2)
[Warning] Server has closed the connection. (index.js, line 2)
```

**Reply by martinpitt** [reply, marked as answer]
Ah, you have a redirect! That's not just accessing the cockpit port with a domain name then, but it is then a reverse proxy. You need to set up `cockpit.conf` with `Origins=`, please see https://cockpit-project.org/guide/latest/cockpit.conf.5.html and https://github.com/cockpit-project/cockpit/wiki/Proxying-Cockpit-over-Apache-with-LetsEncrypt

**Reply by maximehamou** [reply]
@martinpitt I already tried that, but nothing changed!

**Reply by martinpitt** [reply]
Sorry, that needs more logs then -- from apache, from the journal, etc. I don't have further off-hand ideas, I'm afraid.

**Reply by maximehamou** [reply]
I am so sorry! I retried from zero (with just my previous config), and it worked! I think it was either a cache error, or a typing mistake.
Sorry again...

**Reply by martinpitt** [reply]
Hah nice! (Don't worry) :grin:
